packages/transformers/js/src/JSTransformer.js

```diff
@@ -74,9 +76,16 @@ export default new Transformer({
     let ast = asset.ast;
     let code = await asset.getCode();
 
-    // Inline environment variables
-    if (!asset.env.isNode() && (ast.isDirty || ENV_RE.test(code))) {
-      walk.simple(ast.program, envVisitor, {asset, env: options.env});
+    // Inline process/ environment variables
+    if (
+      !asset.env.isNode() &&
+      (ast.isDirty || ENV_RE.test(code) || BROWSER_RE.test(code))
+    ) {
+      walk.ancestor(ast.program, processVisitor, {
```

Comment:
Not sure how the performance is impacted by that change compared to `simple`, but this was the only way I found to do the `isAssignmentExpression` check.

Reply:
The only difference is that it updates an array with `ancestors.push(...)` and `.pop()` before/after calling the visitor.

Reply:
Cool, so I guess this shouldn't have a big impact... :)